ehrql/debugger.py

```diff
@@ -145,7 +145,11 @@ def related_columns_to_records(columns):
 
 
 def related_patient_columns_to_records(columns):
-    for patient_id in columns[0].patient_to_value.keys():
+    patients_all = set()
+    for c in columns:
+        for k in c.patient_to_value.keys():
+            patients_all.add(k)
```

Comment:
I think you're better off using `.update` here e.g
```suggestion
    for c in columns:
        patients_all.update(c.patient_to_value.keys())
```